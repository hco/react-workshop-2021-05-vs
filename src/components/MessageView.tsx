import React from "react";
import Message from "../domain/Message";

const MessageView: React.FunctionComponent<Message> = ({ message, author }) => {
  return (
    <article>
      {message}
      <footer>{author.toUpperCase()}</footer>
    </article>
  );
};

export default MessageView;
