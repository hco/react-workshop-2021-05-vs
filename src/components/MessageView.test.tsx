import { render } from "@testing-library/react";
import Message from "../domain/Message";
import MessageView from "./MessageView";

const testMessage: Message = {
  author: "Hannah",
  date: 1622185877229,
  id: "huodfasiojadsjiodas",
  message: "Bald ist Wochenende!",
};
describe("<MessageView />", () => {
  test("it renders without crashing", () => {
    render(<MessageView message={testMessage} />);
  });
});
