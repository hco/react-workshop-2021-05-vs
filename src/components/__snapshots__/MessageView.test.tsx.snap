// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageView /> it renders according to snapshot 1`] = `
<div>
  <article>
    Bald ist Wochenende!
    <footer>
      HANNAH
    </footer>
  </article>
</div>
`;
